src/Athena/Controllers/api/RequirementController.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+        [HttpPost("api/v1/course/{id}/requirement/satisfies/{reqId}")]
+        public async Task AddSatisfiedRequirementAsync(Course course, Requirement requirement)
+        {
+            if (requirement == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add satisfied requirements {requirement} to course {course} that doesn't exist");
+            }
+            await courses.AddSatisfiedRequirementAsync(course, requirement);
+        }
+
+        [HttpDelete("api/v1/course/{id}/requirement/satisfies/{reqId}")]
+        public async Task RemoveSatisfiedRequirementAsync(Course course, Requirement requirement)
+        {
+            if (requirement == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to remove satisfied requirements {requirement} to course {course} that doesn't exist");
+            }
+            await courses.RemoveSatisfiedRequirementAsync(course, requirement);
+        }
+
+        [HttpPost("api/v1/course/{id}/requirement/prereq/{reqId}")]
+        public async Task AddPrerequisiteAsync(Course course, Requirement prereq)
+        {
+            if (prereq == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add prereq {prereq} to course {course} that doesn't exist");
+            }
+            await courses.AddPrerequisiteAsync(course, prereq);
+        }
+
+        [HttpDelete("api/v1/course/{id}/requirement/prereq/{reqId}")]
+        public async Task RemovePrerequisiteAsync(Course course, Requirement prereq)
+        {
+            if (prereq == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to remove prereq {prereq} to course {course} that doesn't exist");
+            }
+            await courses.RemovePrerequisiteAsync(course, prereq);
+        }
+
+        [HttpPost("api/v1/course/{id}/requirement/prereq/concurrent/{reqId}")]
+        public async Task AddConcurrentPrerequisiteAsync(Course course, Requirement prereq)
+        {
+            if (prereq == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add concurrent prereq {prereq} to course {course} that doesn't exist");
+            }
+            await courses.AddConcurrentPrerequisiteAsync(course, prereq);
+        }
+
+        [HttpDelete("api/v1/course/{id}/requirement/prereq/concurrent/{reqId}")]
+        public async Task RemoveConcurrentPrerequisiteAsync(Course course, Requirement prereq)
+        {
+            if (prereq == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to remove concurrent prereq {prereq} to course {course} that doesn't exist");
+            }
+            await courses.RemoveConcurrentPrerequisiteAsync(course, prereq);
+        }
+
+        [HttpPost("api/v1/program/{id}/requirement")]
```

Comment:
Did you intend to implement this here or in a composite controller? If you intend to implement it here you need to use `/api/v1/student...` to override the controller-wide prefix.